intake_esm/core.py

```diff
@@ -351,142 +472,70 @@ def to_dataset_dict(
+                    total=total, ncols=79, ascii=use_ascii, leave=True, desc='Dataset(s)'
+                )
 
-        if not groupby_attrs:
-            groupby_attrs = self.df.columns.tolist()
+            with concurrent.futures.ThreadPoolExecutor(max_workers=len(sources)) as executor:
+                future_tasks = [executor.submit(_load_source, source) for source in sources]
 
-        # filter groupby_attrs to ensure no columns with all nans
-        def _allnan_or_nonan(column):
-            if self.df[column].isnull().all():
-                return False
-            elif self.df[column].isnull().any():
-                raise ValueError(
-                    f'The data in the {column} column should either be all NaN or there should be no NaNs'
-                )
-            else:
-                return True
+                for i, task in enumerate(concurrent.futures.as_completed(future_tasks)):
+                    ds = task.result()
+                    self._datasets[ds.attrs['intake_esm_dataset_key']] = ds
+                    if self.progressbar:
+                        progress.update(1)
 
-        groupby_attrs = list(filter(_allnan_or_nonan, groupby_attrs))
+                if self.progressbar:
+                    progress.close()
 
-        groups = self.df.groupby(groupby_attrs)
+                return self._datasets
 
-        if agg_columns:
-            keys = '.'.join(groupby_attrs)
-        else:
-            keys = groupby_attrs.copy()
-            keys.remove(path_column_name)
-            keys = '.'.join(keys)
-
-        total = len(groups)
-        dsets = []
-        load_group_dataset_delayed = dask.delayed(_load_group_dataset)
-        tasks = [
-            load_group_dataset_delayed(
-                key,
-                df,
-                self._col_data,
-                agg_columns,
-                aggregation_dict,
-                path_column_name,
-                variable_column_name,
-                use_format_column,
-                mapper_dict,
-                self.zarr_kwargs,
-                self.cdf_kwargs,
-                self.preprocess,
-            )
-            for key, df in groups
-        ]
-
-        client, _is_client_local = _get_dask_client()
-        logger.info(f'Using dask client: {client} --- dask dashboard url: {client.dashboard_link}')
-        futures = client.compute(tasks)
-
-        if self.progressbar:
-            from distributed import progress
-
-            progress(futures)
-            print(
-                f"""\n--> The keys in the returned dictionary of datasets are constructed as follows:\n\t'{keys}'
-                \n--> There is/are {total} group(s)"""
-            )
-
-        dsets = client.gather(futures)
-        self._ds = {group_id: ds for (group_id, ds) in dsets}
-
-        if _is_client_local:
-            client.close()
-            del client
-
-        return self._ds
+    to_xarray = to_dask = to_dataset_dict
```

Comment:
I would lean toward removing `to_xarray` and `to_dask` as methods on the catalog. Other Intake catalogs don't implement these methods AND they really belong on individual data sources.